### ISSUE TITLE
unnecessary_import conflicts with deprecated_export_use

### ISSUE THREAD

**Issue by emersion** [dart-lang/linter#4234]
**Describe the issue**

My code uses `BytesBuilder` from `dart:typed_data`, and other classes from `dart:io`. When importing explicitly `dart:typed_data`, the linter complains:

```
   info • The import of 'dart:typed_data' is unnecessary because all of the used elements are also provided by the import of
          'dart:io' • lib/link_preview.dart:3:8 • unnecessary_import
```

When removing the explicit import, the linter still complains:

```
   info • The ability to import 'BytesBuilder' indirectly is deprecated • lib/link_preview.dart:84:22 • deprecated_export_use
```

**To Reproduce**
Just use `BytesBuilder`.

**Expected behavior**
`unnecessary_import` should not be triggered.

**Comment by srawlins** [comment]
Great catch!
